rofs-filtered.c

```diff
@@ -216,13 +218,19 @@ static void log_regex_error(int error, regex_t *regex, const char* pattern) {
     regfree(regex);
 }
 
+static void add_mode(mode_t mode) {
+    mode &= S_IFMT;
+    for (int i = 0; i < modes_count; ++i)
+        if (mode == modes[i])
+            return;
+    modes[modes_count++] = mode;
```

Comment:
Need to make sure we don't run off the end of the array.

Reply:
only 5 possible entries with room for 8, running off impossible.

Reply:
I can see that. And the code is correct as it stands today. My point is that the 8 is arbitrary, and it's easy to add more than 8 modes and overlook this limit, thus failing by clobbering memory instead of failing by ignoring modes we don't have room for.